client/app/components/alerts/AlertSubscriptions.jsx

```diff
@@ -0,0 +1,154 @@
+        this.props.AlertSubscription.query({ alertId: this.props.alertId }).$promise,
+      ])
+      .then((responses) => {
+        let destinations = responses[0];
+        const subscribers = responses[1];
+
+        const mapF = s => s.destination && s.destination.id;
+        const subscribedDestinations = compact(subscribers.map(mapF));
+
+        const subscribedUsers = compact(subscribers.map(s => !s.destination && s.user.id));
+
+        destinations = destinations.filter(d => !includes(subscribedDestinations, d.id));
+
+        if (!includes(subscribedUsers, this.props.currentUser.id)) {
+          destinations.unshift({
+            name: this.props.currentUser.name + ' (Email)',
+            id: null,
+            icon: 'fa-envelope',
+          });
+        }
+        this.setState({
+          subscribers,
+          destinations,
+        });
+      });
+  }
+
+  state = {
+    selection: null,
+    subscribers: [],
+    destinations: [],
+  }
+
+  saveSubscriber = () => {
+    const sub = new this.props.AlertSubscription({ alert_id: this.props.alertId });
+    if (this.state.selection) {
+      sub.destination_id = this.state.selection;
+    }
+
+    sub.$save(
+      () => {
+        this.props.toastr.success('Subscribed.');
+        const destinations = this.state.destinations.filter(d => d.id !== this.state.selection);
+        this.setState({
+          subscribers: this.state.subscribers.concat(sub),
+          destinations,
+          selection: destinations.length ? destinations[0].id : null,
+        });
+      },
+      () => {
+        this.props.toastr.error('Failed saving subscription.');
+      },
+    );
+  }
+
+  unsubscribe = (subscriber) => {
+    const destination = subscriber.destination;
+    const user = subscriber.user;
+
+    subscriber.$delete(
+      () => {
+        this.props.toastr.success('Unsubscribed');
+        let destinations;
+        const subscribers = without(this.state.subscribers, subscriber);
+        if (destination) {
+          destinations = this.state.destinations.concat(destination);
+        } else if (user.id === this.props.currentUser.id) {
+          destinations = this.state.destinations.concat({
+            id: null,
+            name: this.props.currentUser.name + ' (Email)',
+            icon: 'fa-envelope',
+          });
+        }
+        const newState = { subscribers, destinations };
+        if (destinations.length === 1) {
+          newState.selection = destinations[0].id;
+        }
+        this.setState(newState);
+      },
+      () => {
+        this.props.toastr.error('Failed unsubscribing.');
+      },
+    );
+  }
+
+  updateSelection = selection => this.setState({ selection })
+
+  render() {
+    return (
+      <div className="p-5">
+        <h4>Notifications</h4>
+        <div>
+          <Select
+            value={this.state.selection || null}
+            disabled={this.state.destinations.length === 0}
+            onChange={this.updateSelection}
```

Comment:
I am confused why this would work. `onChange` should pass the event as its default argument so in the function above we'd be setting selection to the event object.

Reply:
Turns out Ant's `Select` component passes the selected value instead of an event.

Reply:
Sounds good...looked odd at a glance.